Bug 1428792 - Remove dashboards from Perfherder

ionutgoldan opened the pull request:
@rwood I simply removed the links to the dashboards we used to have. I left the base code as it was before.

rwood-moz commented on ui/perf.html:
```diff
@@ -35,9 +35,6 @@
                   class="btn btn-view-nav dropdown-toggle">Dashboard
           </button>
           <ul class="dropdown-menu icon-menu" role="menu">
```
Does removing the three menu items (below) automatically remove the "Dashboard" menu item? Or will the "Dashboard" menu item still appear but clicking on it won't do anything? We don't want to just leave an empty "Dashboard" top level menu - that should be hidden. Since we are going to add dashboards in the near future under "Dashboard" is there a way you can just specify that that menu item be hidden?

ionutgoldan replied:
The "Dashboard" menu item is still visible, but has no items to show when clicking it. I can hide it, using an Angular directive.

ionutgoldan commented:
I removed the configuration code, but left the base code, just as you requested. I also removed the old e10s-trend code from the project. Waiting for your feedback.

ionutgoldan commented:
I noticed the API changes around the start_date/end_date/interval param combinations, mentioned [here](https://github.com/mozilla/treeherder/pull/1831/files). Should we revert them?

edmorley commented on ui/js/controllers/perf/dashboard.js:
```diff
@@ -22,19 +22,17 @@ perf.controller('dashCtrl', [
+                    $scope[k] = phDashboardValues[$scope.topic][k];
+                }
+                catch (TypeError) {
+                    $log.debug('"'+ k + '" option not found in ' + '"' + $scope.topic + '" dashboard config');
```
We're moving away from `$log` (see [bug 1428508](https://bugzilla.mozilla.org/show_bug.cgi?id=1428508)), so perhaps remove this entirely or else use `console.debug()`?

ionutgoldan replied:
I'll use console.debug() then. Thanks!

edmorley commented on ui/js/controllers/perf/e10s-trend.js:
```diff
@@ -1,487 +0,0 @@
-    '$state', '$stateParams', '$scope', '$rootScope', '$q', '$httpParamSerializer',
-    'ThRepositoryModel', 'PhSeries', 'PhCompare',
-    'thDefaultRepo', 'phTimeRanges', 'defaultSampleSize',
-    'phComparisonDate', 'defaultBaseDate', 'defaultNewDate', 'phBlockers',
```
`phComparisonDate` is now unused, so can be removed (if ever needed in the future, I'd prefer to import an ES6 style constant rather than use AngularJS `value()`s)

ionutgoldan replied:
Ok

ionutgoldan commented:
I'm closing this so I can make a new PR with squashed commits.
